src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\LanguageConstruct;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class FunctionToConstantFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private static $defaultConfiguration = array(
+        'phpversion' => 'PHP_VERSION',
+        'php_sapi_name' => 'PHP_SAPI',
+        'pi' => 'M_PI',
+    );
+
+    /**
+     * @var array<string, string>
+     */
+    private $configuration;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        $this->configuration = array();
+        foreach ($configuration as $key => $value) {
+            if (!is_string($key) || !array_key_exists($key, self::$defaultConfiguration)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Configuration key must be any of "%s".', implode('", "', array_keys(self::$defaultConfiguration))));
+            }
+
+            if (!is_bool($value)) {
+                throw new InvalidFixerConfigurationException(
+                    $this->getName(),
+                    sprintf(
+                        'Configuration value must be an bool, got "%s" for "%s".',
+                        is_object($value) ? get_class($value) : (gettype($value).'#'.$value),
+                        $key
+                    )
+                );
+            }
+
+            if ($value) {
```

Comment:
and then filter out the `false` values again and for the left `true` values resolve the replacement value? 
The reason why this became odd is because the configuration requires some value. Maybe that is what I don't understand, why can't we just pass list of `strings` any more?